### ISSUE TITLE
vmstorage panic when device has no space left

### ISSUE THREAD

**Issue by Buddruggy** [VictoriaMetrics/VictoriaMetrics#2305]
**Describe the bug**
vmstorage panic when device has no space left

**Expect behavior** 
This vmstorage node should become a read-only node and should not panic.
<img width="1307" alt="image" src="https://user-images.githubusercontent.com/32037815/158100832-67822645-b5d7-4316-9dc1-9aa19de62a71.png">

**Logs**
2022-03-14T02:11:34.823Z        panic   VictoriaMetrics/lib/fs/fs.go:332        FATAL: cannot write 2568 bytes: write /data/data/small/2022_03/tmp/16D8
panic: FATAL: cannot write 2568 bytes: write /data/data/small/2022_03/tmp/16D818AF226CB2B7/index.bin: no space left on device                          
                                                                                                                                                       
goroutine 4302362 [running]:                                                                                                                           
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage({0x8a5e24, 0x20}, {0xc0f6d7b880, 0x76}, 0x2)                                          
        github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:270 +0x994                                                                     
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, {0x8a5e24, 0xa08}, {0x8b2bae, 0x12000}, {0xc0860a15c8, 0xa08, 0x40ac2d}) 
        github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:138 +0x1a5                                                                     
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)                                                                                    
        github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:130                                                                            
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)                                                                                      
        github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:126                                                                            
github.com/VictoriaMetrics/VictoriaMetrics/lib/fs.MustWriteData({0x99bb80, 0xc0b4f7e690}, {0xc0aa3c5500, 0xc87f498000, 0x0})                           
        github.com/VictoriaMetrics/VictoriaMetrics/lib/fs/fs.go:332 +0x108                                                                             
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*blockStreamWriter).flushIndexData(0xc0da556140)                                               
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/block_stream_writer.go:244 +0x125                                                       
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*blockStreamWriter).WriteExternalBlock(0xc0da556140, 0xc1143cc380, 0xc0860a1a00, 0xc0860a1758, 
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/block_stream_writer.go:204 +0x2fb                                                       
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.mergeBlockStreamsInternal(0xc05c84ac80, 0xc108caa000, 0xc05c84ac80, 0x8a8e07, 0x140, 0x17eebb329
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/merge.go:79 +0x799                                                                      
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.mergeBlockStreams(0xc0860a1a00, 0xc0860a1c38, {0xc108caa000, 0xf, 0xf}, 0x2d, 0x40, 0xc01cd4f8a8
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/merge.go:24 +0xf8                                                                       
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*partition).mergeParts(0xc033af4300, {0xc0003cf000, 0xf, 0xc0860a1ed8}, 0x448dcf)              
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/partition.go:1132 +0x8cb                                                                
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*partition).mergePartsOptimal(0xc033af4300, {0xc0003cf000, 0x11, 0x20}, 0x0)                   
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/partition.go:810 +0x114                                                                 
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*partition).flushInmemoryParts(0xc033af4300, {0xc0003cf000, 0x0, 0x0}, 0x0)                    
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/partition.go:792 +0x23a                                                                 
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*partition).inmemoryPartsFlusher(0xc033af4300)                                                 
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/partition.go:764 +0x129                                                                 
github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*partition).startInmemoryPartsFlusher.func1()                                                  
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/partition.go:749 +0x25                                                                  
created by github.com/VictoriaMetrics/VictoriaMetrics/lib/storage.(*partition).startInmemoryPartsFlusher                                               
        github.com/VictoriaMetrics/VictoriaMetrics/lib/storage/partition.go:748 +0x71 

**Version**
```
./usr/bin/vmstorage-prod --version
vmstorage-20211202-134512-tags-v1.70.0-cluster-0-g0f9aaf9ce
```

**Used command-line flags**
`-storageDataPath=/data -retentionPeriod=30d -search.maxUniqueTimeseries=3000000`

**Comment by valyala** [comment]
The enhancement mentioned at https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2305#issuecomment-1072534365 is included in [VictoriaMetrics v1.75.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Now VictoriaMetrics should check for the remaining free disk space every second instead of every 30 seconds. This should significantly reduce the probability of `out of disk space` error for small values for `-storage.minFreeDiskSpaceBytes`.

**Comment by Buddruggy** [comment]
> The enhancement mentioned at [#2305 (comment)](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/2305#issuecomment-1072534365) is included in [VictoriaMetrics v1.75.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Now VictoriaMetrics should check for the remaining free disk space every second instead of every 30 seconds. This should significantly reduce the probability of `out of disk space` error for small values for `-storage.minFreeDiskSpaceBytes`.

OK, thanks

**Comment by hholst80** [comment]
FWIW I think these I/O events (checking TLS certificates etc) should be be subject to configuration options, with defaults.
Hammering the disk every second might not be what I want to do in certain cloud hosted environments.
